src/core/lib/security/security_connector/tls/tls_security_connector.cc

```diff
@@ -202,6 +202,19 @@ void TlsChannelSecurityConnector::check_peer(
                                     : check_arg_->target_name;
       on_peer_checked_ = on_peer_checked;
       gpr_free(peer_pem);
+      const tsi_peer_property* chain =
+        tsi_peer_get_property_by_name(&peer, TSI_X509_PEM_CERT_CHAIN_PROPERTY);
+      if (chain != nullptr) {
+        char* peer_pem_chain =
+          static_cast<char*>(gpr_malloc(chain->value.length + 1));
```

Comment:
s/`gpr_malloc`/`gpr_zalloc` and remove `peer_pem_chain[chain->value.length] = '\0';`

Reply:
Done.